### PR TITLE
Fix all AMS mathord symbols

### PR DESCRIPTION
**Pull request by edemaine**
Here is a fix for #600 (and #609 and #512 and #374).  In fact, all AMS textord symbols were similarly broken.  Here is a complete list of symbols fixed by this PR:

`\hslash,
\triangledown,
\lozenge,
\circledS,
\circledR,
\measuredangle,
\nexists,
\mho,
\Finv,
\Game,
\Bbbk,
\backprime,
\blacktriangle,
\blacktriangledown,
\blacksquare,
\blacklozenge,
\bigstar,
\sphericalangle,
\complement,
\eth,
\diagup,
\diagdown,
\square,
\Box,
\Diamond,
\yen,
\checkmark,
\beth,
\daleth,
\gimel,
\digamma,
\varkappa,
\varnothing,
\maltese`

Basically, in `buildCommon.js`, `mathsym` had logic for choosing between font families `Main-Regular` and `AMS-Regular`, but this applies to types `rel, bin, open, close, inner, punct` only.  `makeOrd` is used specifically for `mathord` and `textord`, and needed similar logic.  To complicate things, `makeOrd` was also losing track of the original macro name and thus AMS vs. Main, replacing it with the unicode symbol.  So I had to refactor code a bit so that `mathDefault` could figure it out.

### CODE REVIEW

**Comment by kevinbarabash** [comment]
@edemaine thanks for the PR.  How did you determine the list of broken chars?  Is it just all mathords?

**Comment by edemaine** [comment]
It is all symbols in symbols.js of font "ams" and type "textord".

File: src/buildCommon.js [hunk 1]
```diff
@@ -116,13 +139,19 @@ var mathit = function(value, mode, options, classes) {
...
             // glyphs for \imath and \jmath do not exist in Math-Italic so we
             // need to use Main-Italic instead
-            utils.contains(mainitLetters, value) ||
-            utils.contains(greekCapitals, value)) {
-        return makeSymbol(
-            value, "Main-Italic", mode, options, classes.concat(["mainit"]));
+            utils.contains(mainitLetters, value)) {
+            // For greek capitals, we have a choice, but Math-Italic
+            // seems closer to LaTeX.
+            //|| utils.contains(greekCapitals, value)) {
```
Comment (kevinbarabash) [inline src/buildCommon.js]: remove commented out code

File: src/buildCommon.js [hunk 2]
```diff
@@ -116,13 +139,19 @@ var mathit = function(value, mode, options, classes) {
...
+        };
     } else {
-        return makeSymbol(
-            value, "Math-Italic", mode, options, classes.concat(["mathit"]));
+        return {
+            fontName: "Math-Italic",
+            fontClass: "mathit"
+        };
     }
 };
```
Comment (kevinbarabash) [inline src/buildCommon.js]: Can you update the doc string for `mathit` now that it no longer creates a symbol?

**Comment by kevinbarabash** [comment]
@edemaine a few minor nits and this'll be ready to merge.  Thanks for fixing all of these symbols.

File: src/buildCommon.js (hunk 1 above)
Comment (edemaine) [inline src/buildCommon.js]: I meant to ask you about this.  Yes, it seems so:

Greek capital letters are in both fonts, `Main-Italic` and `Math-Italic`.  So which should we use? The [master code](https://github.com/Khan/KaTeX/blob/master/src/buildCommon.js#L120) claims to use `Main-Italic` for Greek capital letters.  But if you try the [demo](https://khan.github.io/KaTeX/) with `\mathit{\Omega\imath}`, and inspect, you'll see that the `\Omega` gets class `mathit` while `\imath` gets `mainit`:

```html
<span class="mord mathit" style="margin-right: 0.05017em;">Ω</span>
<span class="mord mainit">ı</span>
```

In fact, the master code ends up calling `mathit()` with a single-unicode-character string (`Ω`), while `greekCapitals` is defined in terms of the macro name (`"\Omega"`).  When my code fixed this, I ended up changing Greek capital letters to use `Main-Italic` font, which caused some tests to fail.  I confirmed with `texcmp` that `Math-Italic` seems slightly closer, so kept it there.

Let me know if you'd like this code changed at all...

File: src/buildCommon.js (hunk 2 above)
Comment (edemaine) [inline src/buildCommon.js]: Fixed, thanks!

**Comment by edemaine** [comment]
Cool!  I just merged with master and moved to ES2016 style.

File: src/buildCommon.js (hunk 1 above)
Comment (kevinbarabash) [inline src/buildCommon.js]: In that case, can you remove the commented out code?  TBH, I'm not sure why we have the same glyphs duplicated in different fonts.  It would be nice if we could avoid this and save a few bytes.  I'm curious to find out why the rendering looks different even though the glyphs are the same.  Maybe the advance or skew metrics for the Greek capitals are differ between fonts.  I think it's safe to investigate this separate from this pull request.

Comment (edemaine) [inline src/buildCommon.js]: Should I remove the list of greekCapitals as well?  It's currently commented out (to prevent lint from complaining).

Comment (kevinbarabash) [inline src/buildCommon.js]: Yes please.

Comment (edemaine) [inline src/buildCommon.js]: Done.

**Comment by kevinbarabash** [comment]
@edemaine thanks for cleaning up the commented out code.  ^_^

**Comment by edemaine** [comment]
No problem!  Now you should be able to close #600, #609, #512, and #374!
